internal/daemon/daemon.go

```diff
@@ -118,14 +119,23 @@ func (d *Daemon) Run(ctx context.Context, listenPort string, stateDir string, so
 		return fmt.Errorf("Failed to initialize directory structure: %w", err)
 	}
 
-	isAlreadyRunning, err := d.os.IsControlSocketPresent()
-	if err != nil {
-		return err
-	}
+	// Clean up the daemon state on an error during init.
+	reverter := revert.New()
+	defer reverter.Fail()
+	reverter.Add(func() {
+		if d.db != nil {
```

Comment:
Can't we run `d.State().Stop()` here instead which sends to the `shutdownDoneCh`?

Reply:
I maybe add the socket cleanup to https://github.com/canonical/microcluster/blob/main/internal/daemon/daemon.go#L83?